src/main/java/SpaceFilter.java

```diff
@@ -0,0 +1,39 @@
+import java.util.ArrayList;
+
+/**
+ * This class represents a filter. It works by checking a given schedule for the given criteria.
+ * It is made "stackable" by returning the schedule if it works, and then that can be used for further calls.
+ */
+public class SpaceFilter implements Filter {
+    // This is the interval between classes that you want, in hours
+    private final int interval;
+
+    public SpaceFilter(int hours) {
+        this.interval = hours;
+    }
+
+    // Essentially, we are returning nothing if the schedule does not pass the filter,
+    // otherwise return the schedule so we can layer
+    @Override
+    public Schedule checkSchedule(Schedule s) {
+        // quick null type check
+        if (s == null) {
+            return null;
+        }
+
+        ArrayList<Timeslot> timeslots = new ArrayList();
+
+        timeslots.addAll(s.getLectures());
+        timeslots.addAll(s.getTutorials());
+
+        for (int i = 0; i < timeslots.size(); i++) {
```

Comment:
I think for optimizing, it would be faster if you could sort this `timeslots` list and then iterate through the list once, checking for space between timeslots? So it's O(n log n) rather than this? I'm not actually sure what this runtime is rn tho lol. is it O(n^2)? Timeslots would need to implement `Comparable` in a way that takes into account weekday as well as the `LocalTime`, but I think that's pretty reasonable.